lib/pheromon_trails.ex

```diff
@@ -0,0 +1,66 @@
+defmodule PheromonTrails do
+  use GenServer
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts[:state], name: opts[:name])
+  end
+
+  def init(%{pheromons: pheromons, nb_ants: nb_ants}) do
+    IO.puts("starting Colony")
+    {:ok, %{
+      pheromons: pheromons,
+      counter: 0 ,
+      evaporation_rate: 0.005 ,
+      nb_ants: nb_ants,
+      best_solution: nil
+      }}
+  end
+
+  def handle_cast({:new_ant_path, path, cost, from}, state) do
```

Comment:
generally speaking it's better to use `handle_call` unless you have a really good reason for it to be a cast, even if you just send back `:ok` as your reply.  1) casts have no delivery guarantee, 2) calls have backpressure responsiveness so data provider genservers get throttled (and possibly even killed, which is a good thing) when their consumers are overwhelmed.

Reply:
@ityonemo Thanks a lot for the comments, it's very helpful ! In this case, I tried first to use handle_call, but it blocked the execution for the other Ant processes. On initialisation, the first one was starting the iteration and since everything was a synchronous recursion afterwards, the other Ant processes were never started. Any thoughts regarding this issue ? I was thinking about triggering the recursion from the PheromonTrails process instead